src/parquet/column/reader.h

```diff
@@ -184,9 +188,9 @@ inline int64_t TypedColumnReader<DType>::ReadBatch(int batch_size, int16_t* def_
   }
 
   // Not present for non-repeated fields
-  if (descr_->max_repetition_level() > 0) {
+  if (descr_->max_repetition_level() > 0 && rep_levels) {
     num_rep_levels = ReadRepetitionLevels(batch_size, rep_levels);
-    if (num_def_levels != num_rep_levels) {
+    if (def_levels && num_def_levels != num_rep_levels) {
```

Comment:
The correct behavior would be to DCHECK that `def_levels` is not null if max_definition_level is > 0 and likewise for max_repetition_level. The only case where the first check will pass and the 2nd does not if if the user calls the API incorrectly. We can add a DCHECK for that, too, if we want (ref_levels passed but not def_levels -- bad API usage)